app/policies/upcoming/read_policy.rb

```diff
@@ -0,0 +1,9 @@
+class Upcoming::ReadPolicy < Struct.new(:user, :read)
```

Comment:
These are all the authorization checks necessary for read access, pretty straightforward.

Reply:
Maybe primarily a Pundit gripe here: I don't understand how this fits in to the rest of the code here. Yes the two methods are very simple, but its magical and somehow lies outside the region scope you have created above.